### PR TITLE
Issue #12716: Make .ci filenames kebab-case

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #12716 

```sh
$ mvn clean compile antrun:run@ant-phase-verify 
```
```
[INFO] --- maven-antrun-plugin:3.1.0:run (ant-phase-verify) @ checkstyle ---
[INFO] Executing tasks
[INFO]      [echo] Checkstyle started (checkstyle_checks.xml): 13/02/2023 06:58:34 PM
[INFO] [checkstyle] Running Checkstyle  on 1496 files
[INFO]      [echo] Checkstyle finished (checkstyle_checks.xml) : 13/02/2023 06:59:37 PM
[INFO]      [echo] Checkstyle started (checkstyle_non_main_files_checks.xml): 13/02/2023 06:58:34 PM
[INFO] [checkstyle] Running Checkstyle  on 3534 files
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/idea_inspection.bat:1: Filename should be in kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/idea_inspection.sh:1: Filename should be in kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/no_old_refs.sh:1: Filename should be in kebab-case. [regexpOnScriptsAndConfigs]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/.ci/sonar_break_build.sh:1: Filename should be in kebab-case. [regexpOnScriptsAndConfigs]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:20 min
[INFO] Finished at: 2023-02-13T18:59:40+01:00

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,14 +178,14 @@ are in the file named "LICENSE.apache20" in this directory.
 [dependabot]:https://dependabot.com
 [dependabot img]:https://api.dependabot.com/badges/status?host=github&repo=checkstyle/checkstyle
 
-[closed issues]:https://github.com/checkstyle/checkstyle/actions/workflows/no_old_refs.yml
-[closed issues img]:https://github.com/checkstyle/checkstyle/actions/workflows/no_old_refs.yml/badge.svg
+[closed issues]:https://github.com/checkstyle/checkstyle/actions/workflows/no-old-refs.yml
```
Comment (stoyanK7) [inline README.md:181]: We forgot this and `run-link-check` in last PR referencing this issue. Now that I am looking, maybe it was better to separate them in a `minor: ..` commit.

File: config/checkstyle_non_main_files_checks.xml [hunk 2]
```diff
@@ -25,8 +25,8 @@
       value="CI folder should only contain executables files"/>
   </module>
   <module name="RegexpOnFilename">
-    <property name="id" value="regexpOnGithubWorkflows"/>
-    <property name="folderPattern" value="[\\/].github[\\/]workflows([\\/]|$)"/>
+    <property name="id" value="regexpOnScriptsAndConfigs"/>
```
Comment (rnveach) [inline config/checkstyle_non_main_files_checks.xml:28]: I am not sure if it is better to have 1 module cover everything, but if we keep this, I would recommend something generic like `kebabCaseFileNaming`.

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml:28]: At first, I went with another check, but they would be identical apart from the `folderPattern`.

I will rename to `kebabCaseFileNaming`.

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Done.

Comment (rnveach) [reply config/checkstyle_non_main_files_checks.xml]: > they would be identical apart from the folderPattern

Yes, but eventually the list may be become too big to maintain. Look at our naming checks in our config. Most of them use the exact same values, but keeping them separate makes it easier to see each identification and add more eventually.

File: README.md (hunk 1 above)
Comment (stoyanK7) [reply README.md]: Done.

File: config/checkstyle_non_main_files_checks.xml (hunk 2 above)
Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: We are adding `config` folder after this PR. Not sure if more are planned to be added after that.

Alright, I will split.

Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Done.

**Review by romani — APPROVED** [review]
Ok to merge, extra check instance is ok.

File: config/checkstyle_non_main_files_checks.xml [hunk 3]
```diff
@@ -25,14 +25,23 @@
...
     <property name="fileExtensions" value="sh, groovy, cmd, bat, yml"/>
     <message key="regexp.filename.mismatch"
              value="Filename should be in kebab-case."/>
   </module>
+  <module name="RegexpOnFilename">
+    <property name="id" value="kebabCaseFileNamingInCiFolder"/>
+    <property name="folderPattern" value="[\\/].ci([\\/]|$)"/>
+    <property name="fileNamePattern" value="^[a-z][a-z0-9]*(\-[a-z0-9]+)*\.[a-z]+$"/>
+    <property name="match" value="false"/>
+    <property name="fileExtensions" value="sh, cmd, bat"/>
```
Comment (rnveach) [inline config/checkstyle_non_main_files_checks.xml:41]: Groovy is missing.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: config/checkstyle_non_main_files_checks.xml (hunk 3 above)
Comment (stoyanK7) [reply config/checkstyle_non_main_files_checks.xml]: Done.

**Review by rnveach — APPROVED** [review]

**Pull request merged**